src/Microsoft.DotNet.Wpf/src/WindowsBase/MS/Internal/IO/Packaging/XmlDigitalSignatureProcessor.cs

```diff
@@ -725,7 +725,10 @@ private SignedXml EnsureXmlSignatureParsed()
             IEnumerable<System.Security.Cryptography.Xml.Reference> objectReferences)
         {
             // don't overwrite
-            Debug.Assert(SignaturePart.GetStream().Length == 0, "Logic Error: Can't sign when signature already exists");
+            using (Stream stream = SignaturePart.GetStream())

```

Comment:
This is a change in behavior.  The Debug.Assert would never evaluate this code before, now it evals even in release.

Reply:
You can pass Func<bool> here and put the entire using inside the assert.

Reply:
You can refactor this into a helper method like this: 

```C#
          [System.Diagnostics.Conditional("DEBUG")]
          void HeperMethod(PackagePart signaturePart)
           {
                   using (Stream stream = SignaturePart.GetStream())
                  {
                     Debug.Assert(stream.Length == 0, "Logic Error: Can't sign when signature already exists");
                   }
            }
```